addons/html_builder/static/src/builder/plugins/overlay_buttons/overlay_buttons_plugin.js

```diff
@@ -0,0 +1,175 @@
+            positionOptions: {
+                position: "top-middle",
+                onPositioned: (overlayEl, position) => {
+                    if (position.top < 0) {
+                        const targetRect = this.target.getBoundingClientRect();
+                        const newTop = targetRect.bottom + 15;
+                        position.top = newTop;
+                        overlayEl.style.top = `${newTop}px`;
+                    }
+                    return;
+                },
+                margin: 15,
+                flip: false,
+            },
+            closeOnPointerdown: false,
+        });
+        this.target = null;
+        this.state = reactive({
+            isVisible: true,
+            buttons: [],
+        });
+
+        this.resizeObserver = new ResizeObserver(() => {
+            this.overlay.updatePosition();
+        });
+
+        // TODO duplicate of builderOverlay => extract somewhere
+        // Recompute the buttons when the window is resized.
+        this.refresh = throttleForAnimation(this.refreshButtons.bind(this));
+        this.addDomListener(window, "resize", this.refresh);
+
+        // On keydown, hide the buttons and then show them again when the mouse
+        // moves.
+        const onMouseMoveOrDown = throttleForAnimation((ev) => {
+            this.toggleVisibility(true);
+            ev.currentTarget.removeEventListener("mousemove", onMouseMoveOrDown);
+            ev.currentTarget.removeEventListener("mousedown", onMouseMoveOrDown);
+        });
+        this.addDomListener(this.editable, "keydown", (ev) => {
+            this.toggleVisibility(false);
+            ev.currentTarget.addEventListener("mousemove", onMouseMoveOrDown);
+            ev.currentTarget.addEventListener("mousedown", onMouseMoveOrDown);
+        });
+
+        // Hide the buttons when scrolling. Show them again when the scroll is
+        // over.
+        const scrollingElement = getScrollingElement(this.document);
+        const scrollingTarget = getScrollingTarget(scrollingElement);
+        this.addDomListener(
+            scrollingTarget,
+            "scroll",
+            throttleForAnimation(() => {
+                this.toggleVisibility(false);
+                clearTimeout(this.scrollingTimeout);
+                this.scrollingTimeout = setTimeout(() => {
+                    this.toggleVisibility(true);
+                }, 250);
+            }),
+            { capture: true }
+        );
+
+        this._cleanups.push(() => {
+            this.removeOverlayButtons();
+            this.resizeObserver.disconnect();
+        });
+    }
+
+    refreshButtons() {
+        if (!this.target) {
+            return;
+        }
+        const buttons = [];
+        this.dispatchTo("get_overlay_buttons", {
+            target: this.target,
+            callback: (buttonList) => buttons.push(...buttonList),
+        });
+        this.state.buttons = buttons;
```

Comment:
for getOverlayButtons of this.getResource("get_overlay_buttons") {
  buttons.push(...getOverlayButtons())
}

Reply:
remove callback get_overlay_buttons just pass a target more simple